### PR TITLE
Add Flex Services Expiration Report

### PR DESCRIPTION
**Pull request by gigxz**
Screen Shot 2022-04-26 at 4 09 35 PM

### CODE REVIEW

File: drivers/health_flexible_service/app/models/health_flexible_service/member_expiration.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    end
+
+    def self.url
+      'health_flexible_service/warehouse_reports/member_expiration'
+    end
+
+    def title
+      'VPR Member Expiration'
+    end
+
```
Comment (awisema) [inline drivers/health_flexible_service/app/models/health_flexible_service/member_expiration.rb]: Do you use any of this, or is it just left over from a report with PDF download?

File: drivers/health_flexible_service/app/models/health_flexible_service/member_expiration.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    def data
+      vpr_t = HealthFlexibleService::Vpr.arel_table
+
+      report_scope_source.
+        select(report_scope_source.column_names + [vpr_t[:end_date]]).
+        where(vpr_t[:open].eq(true).and(vpr_t[:end_date].lteq(Date.today))).
```
Comment (awisema) [inline drivers/health_flexible_service/app/models/health_flexible_service/member_expiration.rb]: This feels like it should be a scope on `HealthFlexibleService::Vpr` to me... Something like `expired_before(date)`?

**Review by awisema — CHANGES_REQUESTED** [review]
Thank you, I think Mike will find this very useful! Just a few style nit-picks...

File: drivers/health_flexible_service/app/models/health_flexible_service/member_expiration.rb (hunk 1 above)
Comment (gigxz) [reply drivers/health_flexible_service/app/models/health_flexible_service/member_expiration.rb]: Hm good question.. the `url` is only used by `viewable_by` above, which I don't believe is used anywhere. I'll remove it.

`title` is used in the xlsx download, I'll update the view to use it as well

**Review by awisema — APPROVED** [review]
Looks good to me, thanks!

File: drivers/health_flexible_service/app/controllers/health_flexible_service/warehouse_reports/member_expiration_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HealthFlexibleService::WarehouseReports
+  class MemberExpirationController < ApplicationController
+    include WarehouseReportAuthorization
+    before_action :require_can_administer_health!
```
Comment (eanders) [inline drivers/health_flexible_service/app/controllers/health_flexible_service/warehouse_reports/member_expiration_controller.rb]: @awisema are we sure this is the correct permission?

Comment (eanders) [inline drivers/health_flexible_service/app/controllers/health_flexible_service/warehouse_reports/member_expiration_controller.rb]: @awisema are we sure this is the correct permission?

File: drivers/health_flexible_service/app/views/health_flexible_service/warehouse_reports/member_expiration/index.haml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+- content_for :title, @report.title
+%h1= content_for :title
+= render '/warehouse_reports/breadcrumbs', tab: 'health-general'
+%div.mt-6 Clients receiving flex services that have expired.
```
Comment (eanders) [inline drivers/health_flexible_service/app/views/health_flexible_service/warehouse_reports/member_expiration/index.haml]: %div is assumed
```suggestion
.mt-6 Clients receiving flex services that have expired.
```

Comment (eanders) [inline drivers/health_flexible_service/app/views/health_flexible_service/warehouse_reports/member_expiration/index.haml]: %div is assumed
```suggestion
.mt-6 Clients receiving flex services that have expired.
```
```suggestion
%div.mt-6 Clients receiving flex services that have expired.
```

**Review by eanders — APPROVED** [review]
Pending confirmation about health permission from @awisema.

File: drivers/health_flexible_service/app/controllers/health_flexible_service/warehouse_reports/member_expiration_controller.rb (hunk 3 above)
Comment (awisema) [reply drivers/health_flexible_service/app/controllers/health_flexible_service/warehouse_reports/member_expiration_controller.rb]: Good question -- is the same permission as the QTR report controller, so I assumed it was fine, but looking at the permissions for Flex Services Administrators, I think it should be `require_can_view_all_vprs!`, and probably `HealthFlexibleService::WarehouseReports::MemberListsController` should be the same?

Comment (eanders) [reply drivers/health_flexible_service/app/controllers/health_flexible_service/warehouse_reports/member_expiration_controller.rb]: That sounds better to me.  @gigxz , can you adjust both of those?

Comment (gigxz) [reply drivers/health_flexible_service/app/controllers/health_flexible_service/warehouse_reports/member_expiration_controller.rb]: Yep, done

**Pull request merged**